qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -472,6 +472,60 @@ def parse_offset_get(s):
         self._buffer2_ready = False
 
         self.connect_message()
+    
+    
+    SNAP_PARAMETERS = {
+            'x': '1',   
+            'y': '2',  
+            'r': '3', 
+            'p': '4',   
+        'phase': '4',  
+           'θ' : '4',
+        'aux 1': '5',  
+        'aux 2': '6',   
+        'aux 3': '7',  
+        'aux 4': '8',  
+         'freq': '9',   
+         'ch 1': '10',
+         'ch 2': '11'  
+    }
+    
+    def snap(self, *parameter_names: str) -> Tuple[float, ...]:
+            """
+            Gets the values of either 2, 3, 4, 5 or 6 parameters at a single instant.
+            For example, SNAP? is a way to query values of
+            X and Y (or R and θ) which are taken at the same time. This is important
+            when the time constant is very short. Using the OUTP? or OUTR? com-
+            mands will result in time delays, which may be greater than the time con-
+            stant, between reading X and Y (or R and θ)
+            Thus reading  X,Y  OR  R,θ yields  a  coherent  snapshot  of  the  output  signal.
+            
```

Comment:
it's great that this corresponds to the description in the manual. However, i think it is also important that the following is also mentioned here:
```
The values of X and Y are recorded at a single instant. The values of R
and θ are also recorded at a single instant. Thus reading X,Y OR R,θ
yields a coherent snapshot of the output signal. If X,Y,R and θ are all
read, then the values of X,Y are recorded approximately 10µs apart from
R,θ. Thus, the values of X and Y may not yield the exact values of R and
θ from a single SNAP? query.
The values of the Aux Inputs may have an uncertainty of up to 32µs. The
frequency is computed only every other period or 40 ms, whichever is
longer. 
The SNAP? command is a query only command. The SNAP? command
is used to
```